tests/functional/test_public_apis.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2012-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import mock
+
+from botocore.session import Session
+from botocore.exceptions import NoCredentialsError
+
+
+class EarlyExit(BaseException):
+    pass
+
+def _test_public_apis_will_not_be_signed(func, kwargs):
+    with mock.patch('botocore.endpoint.Session.send') as _send:
+        _send.side_effect = EarlyExit("we don't care about response here")
+        try:
+            func(**kwargs)
+        except EarlyExit:
+            pass
+        except NoCredentialsError:
+            assert False, "NoCredentialsError should not be triggered"
+        request = _send.call_args[0][0]
+        SigV2_disabled = 'SignatureVersion=2' not in request.url
+        SigV3_disabled = 'X-Amzn-Authorization' not in request.headers
+        SigV4_disabled = 'Authorization' not in request.headers
+        assert SigV2_disabled and SigV3_disabled and SigV4_disabled
+
+def test_cognito_identity_public_apis():
+    session = Session()
+    session.get_credentials = mock.Mock(return_value=None)
+    client = session.create_client('cognito-identity', 'us-east-1')
+    tasks = {
```

Comment:
Rather than have duplicated for each service, why not use the same approach we do for our smoke tests: https://github.com/boto/botocore/blob/develop/tests/integration/test_smoke.py#L18

I'm not really seeing a strong reason why this should deviate from that approach.  I'd prefer to just have one way of doing these types of tests (given a list of service/operation/params, do something and make some assertions about each combo).